### PR TITLE
Create party from playlist

### PR DESCRIPTION
**Pull request by kimberlytze**
Now you can start your party from an existing playlist yay~

@VOjha @vsiri

### CODE REVIEW

**Comment by kimberlytze** [comment]
If someone wants to/knows how to make the dropdown prettier pls help ;-;

File: upnext/create_party.py [hunk 1]
```diff
@@ -15,16 +16,29 @@ def create_party_in_db(request, form):
...
         party.save()
+
+        # Load songs from initial playlist into new playlist
+        if initial_uri != "UpNext - Start a new playlist":
+            tracks_info = sp.user_playlist_tracks(user, playlist_id=initial_uri)["items"]
+            uris = map(lambda x: x["track"]["uri"].split(":")[-1], tracks_info)
+            track_titles = map(lambda x: x["track"]["name"], tracks_info)
+            artists = map(lambda x: x["track"]["artists"][0]["name"], tracks_info)
+            for i in range(len(uris)):
+                if not party.track_set.filter(uri=uris[i]).exists():
```
Comment (VOjha) [inline upnext/create_party.py:47]: Why do you need this check? Is it in case they have duplicate tracks in the playlist?

Comment (kimberlytze) [inline upnext/create_party.py:47]: yep

File: upnext/forms.py [hunk 2]
```diff
@@ -1,10 +1,32 @@
 from django import forms
 from .models import Party
+import spotipy
+
+
+def get_user_playlists(user):
+    user = user.social_auth.get(provider='spotify')
+    token = user.extra_data['access_token']
+    cleaned_playlists = [(unicode("UpNext - Start a new playlist"), unicode("Start a new playlist"))]
```
Comment (VOjha) [inline upnext/forms.py:15]: UNICODE

aaaah

File: upnext/forms.py [hunk 3]
```diff
@@ -1,10 +1,32 @@
...
+    if token:
+        sp = spotipy.Spotify(auth=token)
+        playlists = sp.user_playlists(user)["items"]
+        for p in playlists:
```
Comment (VOjha) [inline upnext/forms.py]: lol can we pls not name a variable `p`

Edit: although we do have `sp` all over the place ugh ;_;

Comment (kimberlytze) [inline upnext/forms.py]: LOL. i'll change it to `playlist`

File: upnext/templates/create.jade [hunk 4]
```diff
@@ -6,5 +6,69 @@ block content
...
+        | {% endfor %}
+    | {% else %}
+    p.required
+      label(for='id_uri') Load songs from any of your playlists
+      select#id_uri(name='uri')
+        | {% for playlist in form.uri.field.choices %}
+        option(value="{{playlist.0}}") {{playlist.1}}
+        | {% endfor %}
+    | {% endif %}
+
```
Comment (VOjha) [inline upnext/templates/create.jade:73]: tl;dr you're a Jade pro

File: upnext/create_party.py (hunk 1 above)
Comment (VOjha) [inline upnext/create_party.py:47]: Naisu 👍

**Comment by VOjha** [comment]
How do we feel about putting bits of the About page on the Create page? To explain wtf is going on

**Pull request merged**